tests/baseline/cases/direct-productid.yml

```diff
@@ -4,7 +4,7 @@
 # To regenerate, run test_baseline.py with PUSHSOURCE_UPDATE_BASELINE=1.
 
 # URL of Source to test.
-url: "productid:/dir/greatest-productid-of-all-time?dest=repo1,repo2,repo3"
+url: "productid:{{ src_dir }}/tests/staged/data/simple_productid/dest1/PRODUCTID/some-cert?dest=repo1,repo2,repo3"
```

Comment:
@rohanpm: This change was done to prevent the test failing on `FileNotFoundError` as the new code tries to read any ProductID that has its `src` property set. However it leads me to a question - Is it safe to assume that the ProductID files are available and readable at the time when the PushItem obejcts are being generated?

Reply:
While theoretically backwards-incompatible with some scripts we don't know about, I think it is acceptable with reasoning:

- 'staged' source is the only backend used in practice which produces ProductIdPushItem
- when using that source, you definitely have the files available because you discovered the item by crawling the filesystem in the first place.